lib/src/api/ewelink_service.dart

```diff
@@ -193,7 +193,25 @@ class EwelinkService {
     return EwelinkDevice.fromJson(response);
   }
 
-  Future<Map<String, dynamic>> _makeRequest({
+  Future<List<EwelinkDevice>> getDevices() async {
+    if (credentials == null) {
+      throw EwelinkGenericException('Not logged in');
+    }
+    Map<String, dynamic> queryParameters = {
+      'appid': appId,
+      'ts': _timestamp,
+      'version': apiVersion.toString(),
+      'getTags': '1',
+    };
+    dynamic response = await _makeRequest(
+      urlPath: '/user/device',
+      queryParameters: queryParameters,
+    );
+
+    return EwelinkDevice.fromJsonList(response['devicelist']);
```

Comment:
We are doing Network requests and passing from Json it's better to encapsulate all body with try catch statement

Reply:
I'm sure if will help, because there is nothing to do here - I will have to rethrow so others can catch.
But I can add a better error message.

Is this what you mean?

```
    try {
      return EwelinkDevice.fromJsonList(response['devicelist']);
    } catch (e) {
      throw EwelinkGenericException(
          'Unexpected response from getDevices(). Received response was ${response}. The response should have `deviceList` inside.\nError: ${e.toString()}');
    }
```

Reply:
Yes and for (or inside) `_makeRequest`